storage/immutable_segment.go

```diff
@@ -91,6 +94,29 @@ type immutableSegmentOptions struct {
 	rawDocSourcePath   []string
 	fieldHashFn        hash.StringArrayHashFn
 	fieldRetriever     persist.FieldRetriever
+	clockOptions       clock.Options
+	instrumentOptions  instrument.Options
+}
+
+type immutableSegMetrics struct {
```

Comment:
I think this might be too fine grained for now, perhaps let's keep the metrics at the namespace level and above but not going below the shard level for now. If we see performance issues, we can always add more fine-grained metrics.